main.go

```diff
@@ -1,18 +1,18 @@
 package main
 
 import (
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"log"
 
-	"github.com/birdsean/review-droid/github"
+	"github.com/birdsean/review-droid/comments"
+	"github.com/birdsean/review-droid/github_client"
 	"github.com/birdsean/review-droid/openai"
 	"github.com/birdsean/review-droid/transformer"
```

Comment:
Bug: There are two imports in this file with the path "github.com/google/go-github/v53/github". This will result in an error at compile-time and this should be fixed.
DEBUG INFO
[File] main.go

[Side] LEFT

[Line] 11

[Raw Comment] [- Line 11] Bug: There are two imports in this file with the path "github.com/google/go-github/v53/github". This will result in an error at compile-time and this should be fixed.
[
[Original Code]  a/main.go b/main.go
index 27a516a..2b8d1a8 100644
 package main
 
 import (
4 -	"encoding/json"
 	"fmt"
6 -	"io/ioutil"
 	"log"
 
9 -	"github.com/birdsean/review-droid/github"
7 +	"github.com/birdsean/review-droid/comments"
8 +	"github.com/birdsean/review-droid/github_client"
 	"github.com/birdsean/review-droid/openai"
 	"github.com/birdsean/review-droid/transformer"
11 +	"github.com/google/go-github/v53/github"
 )
 
 func main() {
15 -	client := github.GithubRepoClient{}
15 +	client := github_client.GithubRepoClient{}
 	client.Init()
 
 	prs, err := client.GetPrs()
 
 	// Iterate over each pull request
 	for _, pr := range prs {
25 -		fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
26 -
27 -		fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
28 -		diff, err := client.GetPrDiff(pr)
29 -		if err != nil {
30 -			log.Fatalf("Failed to get raw diff: %v", err)
25 +		comments := reviewPR(pr, client)
26 +		commitId := pr.GetHead().GetSHA()
27 +		for _, comment := range comments {
28 +			// post to github
29 +			fmt.Printf("Posting comment to PR #%d\n", pr.GetNumber())
30 +			ghComment := client.ParsedCommentToGithubComment(comment, commitId)
31 +			err := client.PostComment(pr, ghComment)
32 +			if err != nil {
33 +				log.Fatalf("Failed to post comment: %v", err)
34 +			}
 		}
36 +	}
37 +}
38 +
39 +func reviewPR(pr *github.PullRequest, client github_client.GithubRepoClient) []*comments.Comment {
40 +	fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
41 +
42 +	fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
43 +	diff, err := client.GetPrDiff(pr)
44 +	if err != nil {
45 +		log.Fatalf("Failed to get raw diff: %v", err)
46 +	}
47 +
48 +	diffTransformer := transformer.DiffTransformer{}
49 +	diffTransformer.Transform(diff)
 
33 -		diffTransformer := transformer.DiffTransformer{}
34 -		diffTransformer.Transform(diff)
51 +	fileSegments := diffTransformer.GetFileSegments()
52 +	allComments := []*comments.Comment{}
53 +	// failedComments := []string{}
 
36 -		segments := diffTransformer.GetSegments()
37 -		allComments := []string{}
38 -		fmt.Printf("Getting commments for %d segments\n", len(segments))
55 +	fmt.Printf("Getting comments for %d segments\n", len(fileSegments))
56 +	for filename, segments := range fileSegments {
 		for _, segment := range segments {
 			comment := retrieveComments(segment)
41 -			allComments = append(allComments, comment)
59 +			if comment == nil {
60 +				// failedComments = append(failedComments, segment)
61 +				continue
62 +			}
63 +			zippedComments, err := comments.ZipComment(segment, *comment, filename)
64 +			if err != nil {
65 +				log.Fatalf("Failed to zip comment: %v", err)
66 +			}
67 +			allComments = append(allComments, zippedComments...)
68 +			if len(allComments) >= 1 {
69 +				break
70 +			}
71 +		}
72 +		if len(allComments) >= 1 {
73 +			break
 		}
43 -
44 -		writeResults(allComments)
 	}
76 +
77 +	return allComments
 }
 
48 -func retrieveComments(segment string) string {
80 +func retrieveComments(segment string) *string {
 	openAiClient := openai.OpenAiClient{}
 	openAiClient.Init()
 	completion, err := openAiClient.GetCompletion(segment)
 		fmt.Printf("Failed to get completion: %v\n", err)
 	}
 
56 -	fmt.Println("********************")
57 -	fmt.Println(*completion)
58 -	fmt.Println("********************")
59 -	return *completion
60 -}
61 -
62 -func writeResults(comments []string) {
63 -	fileContents, err := json.Marshal(comments)
64 -	if err != nil {
65 -		log.Fatalf("Failed to marshal comments: %v", err)
66 -	}
67 -	// save fileContents to results.json
68 -	err = ioutil.WriteFile("results.json", fileContents, 0644)
69 -	if err != nil {
70 -		log.Fatalf("Failed to write comments to file: %v", err)
88 +	if completion != nil {
89 +		fmt.Println("********************")
90 +		fmt.Println(*completion)
91 +		fmt.Println("********************")
 	}
93 +	return completion
 }